File: codeChef/practice/easy/buying2.py
from sys import stdin


for i, ln in enumerate(stdin):
    if i:
        if i % 2 == 0:
            _, n = map(int, ln.split())
        else:
            ls = map(int, ln.split())
            i, j = divmod(sum(ls), n)
            print -1 if j >= min(ls) else i

File: codeChef/practice/easy/buying2.c
#include <stdio.h>

int main() {
    int t;
    scanf("%d", &t);

    while(t-- > 0) {
        int n, k;
        scanf("%d %d", &n, &k);
        int s = 0, m, min = 101;
        while(n-- > 0) {
            scanf("%d", &m);
            s += m;
            if (m < min) {
                min = m;
            }
        }

        printf("%d\n", s % k >= min ? -1 : s / k);
    }
}
